numpy/_core/tests/test_strings.py

```diff
@@ -717,6 +717,73 @@ def test_expandtabs_raises_overflow(self, dt):
             np.strings.expandtabs(np.array("\ta\n\tb", dtype=dt), sys.maxsize)
 
 
+@pytest.mark.parametrize("dt", [
+    "S",
+    "U",
+    pytest.param("T", marks=pytest.mark.xfail(
+        reason="SrtingDType support to be implemented in a follow-up commit",
+        strict=True)),
+])
+class TestMethodsWithoutStringDTypeSupport:
+    @pytest.mark.parametrize("buf,width,fillchar,res", [
+        ('abc', 10, ' ', '   abc    '),
+        ('abc', 6, ' ', ' abc  '),
+        ('abc', 3, ' ', 'abc'),
+        ('abc', 2, ' ', 'abc'),
+        ('abc', 10, '*', '***abc****'),
```

Comment:
This looks wrong - left should be longer than right if they're not equal (also above for space).

Though, python seems remarkably inconsistent:
```
In [16]: 'abc'.center(10, '*')
Out[16]: '***abc****'
In [21]: 'abcd'.center(11, '*')
Out[21]: '****abcd***'
```

There must be some logic to this, so never mind my earlier comment about the implementation...